server/src/main/scala/com/cloudera/livy/utils/SparkYarnApp.scala

```diff
@@ -209,16 +211,43 @@ class SparkYarnApp private[utils] (
       listener.foreach(_.appIdKnown(appId.toString))
 
       val pollInterval = SparkYarnApp.getYarnPollInterval(livyConf)
+      var appInfo = AppInfo()
       while (isRunning) {
-        Clock.sleep(pollInterval.toMillis)
-
-        // Refresh application state
-        val appReport = yarnClient.getApplicationReport(appId)
-        yarnDiagnostics = getYarnDiagnostics(appReport)
-        changeState(mapYarnState(
-          appReport.getApplicationId,
-          appReport.getYarnApplicationState,
-          appReport.getFinalApplicationStatus))
+        try {
+          Clock.sleep(pollInterval.toMillis)
+
+          // Refresh application state
+          val appReport = yarnClient.getApplicationReport(appId)
+          yarnDiagnostics = getYarnDiagnostics(appReport)
+          changeState(mapYarnState(
+            appReport.getApplicationId,
+            appReport.getYarnApplicationState,
+            appReport.getFinalApplicationStatus))
+
+          val latestAppInfo = {
+            val attempt =
+              yarnClient.getApplicationAttemptReport(appReport.getCurrentApplicationAttemptId)
+            val driverLogUrl =
+              Try(yarnClient.getContainerReport(attempt.getAMContainerId).getLogUrl)
```

Comment:
You are right and Livy officially supports yarn-cluster mode and local mode only.